koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -859,6 +862,31 @@ def checkLabels(self, src, build_tag, label_overwrites=None):
             return (data['COMPONENT'], labels_wrapper.get_expected_nvr())
         return (data['COMPONENT'], None)
 
+    def check_servers_up(self, flags=None):
```

Comment:
@mlangsdorf, did you have a chance to look into the checkHost function? AFAIU, it's a callback that the koji builder will automatically call when deciding to take a task. https://pagure.io/koji/blob/master/f/koji/daemon.py#_1268

The benefit of using this approach is that the task remains in the free state so it's clear that it's actually running.

This needs further investigation because I'm not sure how often `checkHost` is called, and how often it's called. Trying this out in osbs-box would be ideal.